packages/paste-core/components/truncate/src/index.tsx

```diff
@@ -1,12 +1,33 @@
-import {styled} from '@twilio-paste/styling-library';
-
-const Truncate = styled.span`
-  display: inline-block;
-  max-width: 100%;
-  overflow: hidden;
-  text-overflow: ellipsis;
-  white-space: nowrap;
-`;
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {safelySpreadBoxProps, Box, BoxProps} from '@twilio-paste/box';
+
+export interface TruncateProps extends Pick<BoxProps, 'as'> {
+  children: NonNullable<React.ReactNode>;
+}
+
+const Truncate = React.forwardRef<HTMLSpanElement, TruncateProps>(({children, ...props}, ref) => {
+  return (
+    <Box
+      {...safelySpreadBoxProps(props)}
```

Comment:
Do we care to safelySpread here? Truncate isn't really a designed component so it may be fine to just spread all Box props.. What do you think?

Reply:
Yeah I don't think we need it here either. We probably don't need to be concerned with what gets passed here since there aren't any design implications.